users/views.py

```diff
@@ -60,23 +61,29 @@ def post(self, request):
 
                 token = RefreshToken.for_user(custom_user).access_token
                 current_site = get_current_site(request).domain
-                relative_link = reverse('send_email_confirmation')
-                abs_url = 'http://' + current_site + relative_link + '?token=' + str(token)
+                relative_link = reverse('verify-email', kwargs={'token': token, 'user_id': custom_user.id})
+                abs_url = 'http://'+ current_site + relative_link + '?token=' + str(token) + '?id=' + str(custom_user.id) 
                 email_body = 'Hi ' + custom_user.first_name + ' Use the link below to verify your email \n' + abs_url
-                sended_data = {'email_body': email_body, 'email_subject': 'Email confirmation',
-                               'to_email': custom_user.email}
-                # Util.send_email(data=sended_data)
-
-                return Response({"message": "User created successfully"}, status=status.HTTP_201_CREATED)
+                sended_data = {'email_body': email_body, 'email_subject': 'Email confirmation', 'to_email': custom_user.email}
+                Util.send_email(data=sended_data)
+                                
+                return Response({"User id": custom_user.id, "User name": custom_user.first_name, "message": "User created successfully"}, status=status.HTTP_201_CREATED)
             else:
                 return Response({"message": "Failed to create user"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-class VerifyEmailAPIView(APIView):
-    def get(self):
-        pass
+class SendEmailConfirmationAPIView(APIView):
+    def get(self, request, token=None, user_id=None):
+        if not token or not user_id:
+            return Response({"message": "Token or user ID is invalid"}, status=status.HTTP_400_BAD_REQUEST)
+        
+        user = get_object_or_404(CustomUser, id=user_id)
+        user.is_email_valid = True
+        user.save()
+        
+        return Response({"User name":user.first_name, "email": user.email, "message": "Email verified successfully"}, status=status.HTTP_200_OK)
```

Comment:
Your method checks if the `token` and `user_id` are provided but does not actually validate the token against any stored or expected value. It's crucial to verify that the token is valid and matches what was sent to the user's email, ensuring that it's not expired or tampered with.
You may Implement a function to validate the token, e.g.
```
def validate_token(token, user):
    # Paste your logic to validate the token
    return token == user.expected_token and not token_expired(token)
```

Reply:
After searching about token validation cheated this check:
if decoded_token.get('user_id') != int(user_id):
            return Response({'error': 'Invalid token'}, status=status.HTTP_400_BAD_REQUEST)